### PR TITLE
Start Posts Model

### PR DESCRIPTION
**Pull request by chaitali**

### CODE REVIEW

**Comment by chaitali** [comment]
Tests failing on PR because of sqlite errors...... I assume they'll disappear when we change the db to postgres.

File: core/settings.py [hunk 1]
```diff
@@ -79,8 +80,12 @@
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'ENGINE': 'django.db.backends.postgresql',
```
Comment (cmackenzie1) [inline core/settings.py]: We should remove all the extra info (ie, username and password) and use the environment variable `DATABASE_URL`.

Comment (chaitali) [reply core/settings.py]: so `DATABASE_URL = 'postgres://cmput404:abramiscool123!@cmput404-dev.c1dsguk3kuvt.us-west-2.rds.amazonaws.com:5432'` and then `DATABASE = {}` then `DATABASE['default'] = DATABASE_URL`? Unsure how to implement this :P

Comment (cmackenzie1) [reply core/settings.py]: I would add this package: https://django-environ.readthedocs.io/en/latest/

Then do something like this:

```python
# settings.py
import environ

ROOT_DIR = environ.Path(__file__) - 3  # (homefinder/config/settings/base.py - 3 = homefinder/)
APPS_DIR = ROOT_DIR.path('homefinder')

env = environ.Env()

READ_DOT_ENV_FILE = env.bool('DJANGO_READ_DOT_ENV_FILE', default=False)
if READ_DOT_ENV_FILE:
    # OS environment variables take precedence over variables from .env
    env.read_env(str(ROOT_DIR.path('.env')))

# DATABASES
# ------------------------------------------------------------------------------
# https://docs.djangoproject.com/en/dev/ref/settings/#databases

DATABASES = {
    'default': env.db('DATABASE_URL', default='postgres://cmput404:abramiscool123!@cmput404-dev.c1dsguk3kuvt.us-west-2.rds.amazonaws.com:5432'),
}
```
```bash
export DATABASE_URL=postgres://cmput404:abramiscool123!@cmput404-dev.c1dsguk3kuvt.us-west-2.rds.amazonaws.com:5432``

Comment (cmackenzie1) [reply core/settings.py]: I used my CMPUT 401 example, but we jsut need to change the urls (and maybe that `ROOT_DIR` thing)

Comment (chaitali) [reply core/settings.py]: `ROOT_DIR = environ.Path(__file__) - 2  # (cmput404-project/core/settings.py - 2 = cmput404-project/)
APPS_DIR = ROOT_DIR.path('cmput404-project')

env = environ.Env()

READ_DOT_ENV_FILE = env.bool('DJANGO_READ_DOT_ENV_FILE', default=False)
if READ_DOT_ENV_FILE:
    # OS environment variables take precedence over variables from .env
    env.read_env(str(ROOT_DIR.path('.env')))`

Did this and I'm assuming I have to add a .env file in the main directory, with the export? Did this and my tests aren't passing :\

File: core/posts/models.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        default='PUBLIC'
+    )
+
+    # Optional Fields
+    description = models.CharField(max_length=100, blank=True, null=True)
+    categories = ArrayField(models.CharField(max_length=200), default=list)
+
+class Comments(CommonData):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    content_type = models.CharField(max_length=30, default="text/markdown")
```
Comment (cmackenzie1) [inline core/posts/models.py]: You set a default for the comments - does Posts also need this specified?

Comment (chaitali) [reply core/posts/models.py]: So the content_type for comments is looks like it defaults to markdown, which is why I defaulted it like that.

**Pull request merged**